### PR TITLE
Mask sensitive information

### PR DESCRIPTION
**Pull request by bustikiller**
We need to mask some sensitive information such as credentials and API tokens.

### CODE REVIEW

**Comment by trusche** [comment]
Hi @bustikiller, thanks for the PR, and that's a cool feature in principle. 

I'll have to spend some time reviewing it and thinking through how this would cover the most common use cases - this implementation, at a glance, would only filter out key=value pairs, and for example not work on quoted value strings or JSON. If you want to take a shot at that, go ahead, otherwise I'll take it from here, but it will take a bit to get merged then.

File: lib/httplog/http_log.rb [hunk 1]
```diff
@@ -95,14 +96,33 @@ def log_body(body, encoding = nil, content_type = nil)
...
+        log_data_lines(filter_out(data))
       else
-        log("Response:\n#{data}")
+        log("Response:\n#{filter_out(data)}")
       end
     rescue BodyParsingError => e
       log("Response: #{e.message}")
     end
 
+    def filter_out(msg)
```
Comment (trusche) [inline lib/httplog/http_log.rb]: I'd add a guard clause here so we don't do any of the work below if there aren't any keywords to filter.

Comment (trusche) [inline lib/httplog/http_log.rb]: IMO, a better name for this method would describe the returned result - e.g. `masked_data`.

**Comment by bustikiller** [comment]
@trusche thanks for the fast feedback. 

I will try to find some time to make this work with keywords inside JSON. However, I don't understand what you mean by "will not work on quoted string values". Could you please provide an example?

Thanks

**Comment by bustikiller** [comment]
@trusche I added tests with some JSON-body examples that came to my mind and fixed the `masked_data` method accordingly.

**Comment by trusche** [comment]
Cool, thanks. I'll tweak this a bit, should go live shortly.

**Pull request merged**